java/src/jmri/jmrit/throttle/ThrottlesPreferencesControlsSettingsPane.java

```diff
@@ -0,0 +1,482 @@
+            tpwkc.setNextThrottleWindowKeys(tfNextThrottleWindow.getShortCuts() );
+        }
+        if (tfPrevThrottleWindow.isDirty()) {
+            tpwkc.setPrevThrottleWindowKeys(tfPrevThrottleWindow.getShortCuts() );
+        }
+        if (tfNextThrottleFrame.isDirty()) {
+            tpwkc.setNextTrottleFrameKeys( tfNextThrottleFrame.getShortCuts() );
+        }
+        if (tfPrevThrottleFrame.isDirty()) {
+            tpwkc.setPrevThrottleFrameKeys( tfPrevThrottleFrame.getShortCuts() );
+        }
+        if (tfNextRunningThrottleFrame.isDirty()) {
+            tpwkc.setNextRunThrottleFrameKeys( tfNextRunningThrottleFrame.getShortCuts() );
+        }
+        if (tfPrevRunningThrottleFrame.isDirty()) {
+            tpwkc.setPrevRunThrottleFrameKeys( tfPrevRunningThrottleFrame.getShortCuts() );
+        }
+        if (tfNextThrottleInternalWindow.isDirty()) {
+            tpwkc.setNextThrottleInternalWindowKeys( tfNextThrottleInternalWindow.getShortCuts() );
+        }
+        if (tfPrevThrottleInternalWindow.isDirty()) {
+            tpwkc.setPrevThrottleInternalWindowKeys( tfPrevThrottleInternalWindow.getShortCuts() );
+        }
+        if (tfGotoControl.isDirty()) {
+            tpwkc.setMoveToControlPanelKeys( tfGotoControl.getShortCuts() );
+        }
+        if (tfGotoFunctions.isDirty()) {
+            tpwkc.setMoveToFunctionPanelKeys( tfGotoFunctions.getShortCuts() );
+        }
+        if (tfGotoAddress.isDirty()) {
+            tpwkc.setMoveToAddressPanelKeys( tfGotoAddress.getShortCuts() );
+        }
+        if (tfForward.isDirty()) {
+            tpwkc.setForwardKeys( tfForward.getShortCuts() );
+        }
+        if (tfReverse.isDirty()) {
+            tpwkc.setReverseKeys( tfReverse.getShortCuts() );
+        }
+        if (tfSwitchDir.isDirty()) {
+            tpwkc.setSwitchDirectionKeys( tfSwitchDir.getShortCuts() );
+        }
+        if (tfSpeedIdle.isDirty()) {
+            tpwkc.setIdleKeys( tfSpeedIdle.getShortCuts() );
+        }
+        if (tfSpeedStop.isDirty()) {
+            tpwkc.setStopKeys( tfSpeedStop.getShortCuts() );
+        }
+        if (tfSpeedUp.isDirty()) {
+            tpwkc.setAccelerateKeys( tfSpeedUp.getShortCuts() );
+        }
+        if (tfSpeedDown.isDirty()) {
+            tpwkc.setDecelerateKeys( tfSpeedDown.getShortCuts() );
+        }
+        if (tfSpeedUpMore.isDirty()) {
+            tpwkc.setAccelerateMoreKeys( tfSpeedUpMore.getShortCuts() );
+        }
+        if (tfSpeedDownMore.isDirty()) {
+            tpwkc.setDecelerateMoreKeys( tfSpeedDownMore.getShortCuts() );
+        }
+        for (int i=0; i<tfFunctionKeys.length; i++) {
+            if (tfFunctionKeys[i].isDirty) {
+                tpwkc.setFunctionsKeys (i, tfFunctionKeys[i].getShortCuts() );
+            }
+        }
+        try {
+            float sm = Float.parseFloat(tfSpeedMultiplier.getText());            
+            if (Float.compare(sm,  tpwkc.getMoreSpeedMultiplier())!=0) {
```

Comment:
This might work but is a bit risky. If I remember correctly, `0.1 * 10` is not equal to `1`. And `(1/3) * 3` is not equal to `1`.

The problem is that computers have difficult to store floating numbers exactly, so `(1/3) * 3` is `0.99999999` or something like that. And `0.1` is not stored exactly as 0.1 but something like 0.099999999. It was a long time ago I was in school and learned these things, so the details might be wrong, but the point is that you can't trust floating numbers too much.

Often it's better to do a check like:

`if (Math.abs(sm - tpwkc.getMoreSpeedMultiplier()) < 0.00001) {`

That allows for some rounding errors.

Reply:
it was my understanding that this Float.compare() was doing exactly that. Anyway reworked as asked.